### PR TITLE
Add e2e tests for typical contracts #918

### PR DESCRIPTION
**Pull request by lzxm160**
add array-delete.sol test for #918

### CODE REVIEW

File: tools/executiontester/blockchain/typedef.go [hunk 1]
```diff
@@ -7,12 +7,16 @@
 package blockchain
 
 const (
-	// Erc721Binary a simple erc721 token bin
+	// Erc721Binary a simple erc721 token bin,the following is its method
```
Comment (lizhefeng) [inline tools/executiontester/blockchain/typedef.go]: Please add a space after the comma.

File: tools/executiontester/blockchain/typedef.go [hunk 2]
```diff
@@ -7,12 +7,16 @@
...
 	// CreateTo createTo(address _owner,uint256 _tokenId)
 	CreateTo = "fdb05e85"
 	// TransferFrom transferFrom(address _from, address _to, uint256 _tokenId)
 	TransferFrom = "23b872dd"
 	// BalanceOf BalanceOf(opts *bind.CallOpts, _owner common.Address)
 	BalanceOf = "70a08231"
+	// ArrayDeleteBin is a simple test contract,details in array-delete.sol,the following is its method
```
Comment (lizhefeng) [inline tools/executiontester/blockchain/typedef.go]: Ditto

File: tools/executiontester/blockchain/array-delete.go [hunk 3]
```diff
@@ -0,0 +1,45 @@
...
+	}
+)
+
+// NewArrayDelete creates a new ArrayDelete contract
+func NewArrayDelete(exp string) ArrayDelete {
+	return &arrayDelete{Contract: NewContract(exp)}
+}
+
+// MainFunc is function main() returns (uint[])
+func (f *arrayDelete) MainFunc() (ret []*big.Int, err error) {
```
Comment (dustinxie) [inline tools/executiontester/blockchain/array-delete.go]: GetArray()

File: tools/executiontester/blockchain/typedef.go (hunk 1 above)
Comment (dustinxie) [inline tools/executiontester/blockchain/typedef.go]: simple erc721 token, the following is its binary

File: tools/executiontester/blockchain/typedef.go (hunk 2 above)
Comment (dustinxie) [inline tools/executiontester/blockchain/typedef.go]: test contract, source code in array-delete.sol, the following is its binary

**Pull request merged**